src/ast/ast_build_reference_map.c

```diff
@@ -334,11 +334,17 @@ static void _AST_MapForeachClauseReferences
 	ASSERT(ast != NULL);
 	ASSERT(foreach_clause != NULL);
 
+	// map the alias of the list component
+	const cypher_astnode_t *ident =
+		cypher_ast_foreach_get_identifier(foreach_clause);
+	const char *identifier_name = cypher_ast_identifier_get_name(ident);
+	_AST_UpdateRefMap(ast, identifier_name);
+
```

Comment:
Please remove this logic.
The loop variable e.g. `v` isn't referenced, within the loop header, it might be referenced within the body, but we don't really care about that, as we're guarantee the `v` will be part of our record mapping.